minimal.js

```diff
@@ -93,9 +93,22 @@ module.exports = function (dirname, keys, opts) {
 
   var log = OffsetLog(path.join(dirname, 'log.offset'), {blockSize:1024*16, codec:codec})
 
-  //NOTE: must use db.ready.set(true) at when migration is complete
+  const maps = []
+  const chainMaps = (val, cb) => {
+    let idx = -1 // haven't entered the chain yet
+    const next = (err, val) => {
+      idx += 1
+      if (err || idx === maps.length)
+        cb(err, val)
+      else
+        maps[idx](val, next)
+    }
+    next(null, val)
+  }
```

Comment:
There's also a cleaner implementation if you're open to having [`run-waterfall`](https://github.com/feross/run-waterfall) added to the project:

```js
function chainMaps(value, done) {
  waterfall([
    cb => cb(null, value),
    ...maps,
    done
  ]
}
```

Is this worth adding another package, or would you prefer to keep the current implementation?

Suggested here: `%RSVrlUQcZSmkmXOqMSfr0XTKeCHnFPHyWkKmU5LKSaY=.sha256`